### PR TITLE
Reactivate legacy unicode test, test2 and add Expecto tests to build pipeline

### PR DESCRIPTION
**Pull request by gdziadkiewicz**
Based on the content of `build.bat` and `run.bat` files left after legacy tests I was able to find out how to use some of the legacy tests marked as incomplete (I was mostly interested in Unicode tests). While working on it I managed to get rid of the distinction between `main` and `main-unicode` and to remove usages of ml and some of the OCaml adapters. 

As for the Expecto tests I decided to go with `YoloDev.Expecto.TestSdk` because it allowed to easily run the tests in the pipeline via `dotnet test` and as a bonus allows to run them in VS:
image

### CODE REVIEW

**Comment by baronfel** [comment]
I love the addition of the test suite! Looking forward to seeing this green :)

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-latest, windows-latest, macOS-latest]
-        dotnet: [5.0.100]
+        dotnet: [5.0.404]
```
Comment (gdziadkiewicz) [inline .github/workflows/main.yml:18]: Needed to change it to fix https://github.com/dotnet/core/issues/6595#issuecomment-905086918

File: tests/FsYacc.Core.Tests/Sample.fs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿module FsYacc.Core.Tests.Sample
+
+open Expecto
+
+[<Tests>]
+let tests =
+    test "sample" {
```
Comment (gdziadkiewicz) [inline tests/FsYacc.Core.Tests/Sample.fs:7]: I had trouble with quickly coming up with a meaningful test for the fsyacc part so I decided to go with a sample to leave it "ready" for real tests in the future.

File: tests/fsyacc/Test1/test1.fsproj [hunk 3]
```diff
@@ -3,18 +3,16 @@
...
-    <OtherFlags>--mlcompatibility</OtherFlags>
+    <NoWarn>FS0760</NoWarn>
   </PropertyGroup>
   <ItemGroup>
-    <Compile Include="../lexing.fs" />
-    <Compile Include="../parsing.fs" />
     <Compile Include="../../../src/Common/Arg.fs" />
-    <Compile Include="../arg.fs" />
-    <Compile Include="../tree.ml" />
+    <Compile Include="../arg2.fs" />
```
Comment (gdziadkiewicz) [inline tests/fsyacc/Test1/test1.fsproj:10]: Funny finding - two files with the same name did compile fine but displayed incorrectly(only one of them was displayed in solution explorer) in VS2022. I decided to rename the one form tests folder to `arg2` to make it easier to play with the tests projects in VS2022.

File: tests/fsyacc/main.fs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        use is = File.OpenRead filename
+        let lexbuf = createLexBuffer <@ TestLexer.token @> is
+        if !tokenize then
+          while true do 
+            Printf.eprintf "tokenize - getting one token\n"
+            let t = TestLexer.token lexbuf
+            Printf.eprintf "tokenize - got %s, now at char %d\n" (TestParser.token_to_string t) (lexbuf.StartPos).pos_cnum
+            match t with 
+            | TestParser.EOF -> exit 0
+            | TestParser.IDENT s ->
```
Comment (gdziadkiewicz) [inline tests/fsyacc/main.fs:39]: It's hard to see in the diff but this `match` in the `main.fs` (originally `main.ml`) comes from `main-unicode.ml` and allows testing the exact names of the lexed IDENT tokens.

**Comment by gdziadkiewicz** [comment]
I believe that this is ready to be reviewed and merged if there are no issues.

**Comment by gdziadkiewicz** [comment]
@dsyme would you find a moment to review?

**Comment by gdziadkiewicz** [comment]
@baronfel @kkm000 are you perhaps able to review?

**Comment by dsyme** [comment]
Looks good to me!

**Pull request merged**